### ISSUE TITLE
Changelog Mirror 

### ISSUE THREAD

**Issue by snakers4** [snakers4/silero-vad#2]
Just a handy issue to be notified of latest changes and micro-releases (we will mostly changing the models)

**Comment by snakers4** [comment]
Initial models, examples, utils for VAD only uploaded (no number detector or language classifier yet)

**Comment by snakers4** [comment]
First readable public release

**Comment by snakers4** [comment]
Added VAD latency and throughput metrics

**Comment by snakers4** [comment]
Updated VAD quality
Before / after (precision / recall)
![image](https://user-images.githubusercontent.com/12515440/102921146-ef946980-449c-11eb-8d48-1263add03932.png)

**Comment by adamnsandle** [comment]
Added < 250ms compatibility
![image](https://user-images.githubusercontent.com/36505480/103082985-05607680-45e4-11eb-937b-07cf88b77dc4.png)

**Comment by Sontref** [comment]
Added number detector

**Comment by snakers4** [comment]
Number detector example, readme update + FAQ

**Comment by snakers4** [comment]
Added a utility to tune the VAD params properly for a domain

**Comment by snakers4** [comment]
Some final benchmarks posted here - https://github.com/pyannote/pyannote-audio/issues/604#issue-798003383
Probably we are done with benchmarks for now

**Comment by snakers4** [comment]
Added micro (10k params, 100x smaller) VAD models

**Comment by snakers4** [comment]
Added micro (10k params, 100x smaller) VAD models for 8 kHz audio

**Comment by snakers4** [comment]
- Added mini (100k params) VAD models for 8 kHz and 16 kHz 
- Added adaptive vad iterator

https://github.com/snakers4/silero-vad/pull/54

**Comment by snakers4** [comment]
- Fixed examples and notebooks
- Updated README
- Added adaptive examples

**Comment by snakers4** [comment]
- Added a language classifier for 116 languages
- It classifies audios into languages and mutually intelligible language groups (i.e. Serbian + Bosnian + Croatian, Russian + Ukranian + others, Hindi + Urdu, etc), see the full list [here](https://raw.githubusercontent.com/snakers4/silero-vad/master/files/lang_group_dict_116.json) and [here](https://raw.githubusercontent.com/snakers4/silero-vad/master/files/lang_dict_116.json)
- Probably some artifical / unspoken languages will be excluded and a large model will be trained

**Comment by snakers4** [comment]
improved language classifier

- 95 languages (85% accuracy), 58 language groups (90% accuracy)
- Mutually intelligible languages are united into language groups  (i.e. Serbian + Croatian + Bosnian are very similar)
- Trained on approx 20k hours of data (10k of which are for 5 most popular languages)
- 4.7M params

**Comment by snakers4** [comment]
# New V3 Silero VAD is Already Here

## Main changes

- One VAD to rule them all! New model includes the functionality of the previous ones with [improved quality](https://github.com/snakers4/silero-vad/wiki/Quality-Metrics) and [speed](https://github.com/snakers4/silero-vad/wiki/Performance-Metrics)!
- Flexible sampling rate, `8000 Hz` and `16000 Hz` are supported;
- Flexible chunk size, minimum chunk size is just 30 milliseconds!
- 100k parameters;
- GPU and batching are supported;
- Radically simplified examples;

## Migration

Please see the new [examples](https://github.com/snakers4/silero-vad/wiki/Examples-and-Dependencies#examples).

New `get_speech_timestamps` is a simplified and unified version of the old deprecated `get_speech_ts` or `get_speech_ts_adaptive` methods.

```
speech_timestamps = get_speech_timestamps(wav, model, sampling_rate=16000)
```

New `VADIterator` class serves as an example for streaming tasks instead of old deprecated `VADiterator` and `VADiteratorAdaptive`.

```
vad_iterator = VADIterator(model)
window_size_samples = 1536

for i in range(0, len(wav), window_size_samples):
   speech_dict = vad_iterator(wav[i: i+ window_size_samples], return_seconds=True)
   if speech_dict:
       print(speech_dict, end=' ')
vad_iterator.reset_states()

```

**Comment by snakers4** [comment]
# Even Better V3 Silero VAD

- Models with even higher quality (just see the plots with metrics!);
- New model ~ large model >> all previous (even large) models;
- Now model works properly quality-wise, i.e. 100ms >  60ms > 30ms and16 kHz > 8 kHz;

**Comment by snakers4** [comment]
This summarises new progress well

![image](https://user-images.githubusercontent.com/12515440/145581727-870b43f9-feae-4cde-9702-aa8d3438e379.png)

**Comment by snakers4** [comment]
# New V3 ONNX VAD Released 

**We finally were able to port a model to ONNX:**

- Compact model (~100k params);
- Both PyTorch and ONNX models are not quantized;
- Same quality model as the latest best PyTorch release;
- Only 16kHz available now (ONNX has some issues with if-statements and / or tracing vs scripting) with cryptic errors;
- In our tests, on short audios (chunks) ONNX is 2-3x faster than PyTorch (this is mitigated with larger batches or long audios);
- Audio examples and non-core models moved out of the repo to save space;

**Comment by snakers4** [comment]
# Support For Sampling Rates Higher Than 16 kHz

- `jit` model now can handle 8, 16, 32 and 48 kHz directly (change implemented within the model itself);
- `onnx` model as well, but only via external wrappers (we just use each n-th sample for higher sampling rates);
- This support is mostly a hack, i.e. we just use each n-th sample for higher sampling rates (instead of averaging);

**Comment by snakers4** [comment]
# ⚠️ Important Information for VAD Python Users ⚠️

If you are using the VAD in a:

- multi-threaded or
- a multi-process application

**Do not forget to disable gradients in EACH process and / or thread.**
Otherwise memory may leak noticeably.

**Comment by snakers4** [comment]
![image](https://user-images.githubusercontent.com/12515440/155746169-251b4929-8a94-4687-a432-c6f2f5ccca61.png)

![image](https://user-images.githubusercontent.com/12515440/155746188-0ee13075-f0a8-4e4e-a73b-d55ecd6d1bbc.png)

**Comment by adamnsandle** [comment]
# New V4 VAD Released 

Changes:
- Improved [quality](https://github.com/snakers4/silero-vad/wiki/Quality-Metrics#silero-vad-vs-old-silero-vad)
- Improved [perfomance](https://github.com/snakers4/silero-vad/wiki/Performance-Metrics#silero-vad-performance-metrics)
- Both 8k and 16k sampling rates are now supported by the ONNX model
- Batching is now supported by the ONNX model
- Added `audio_forward` method for one-line processing of a single or multiple audio without postprocessing

**Comment by snakers4** [comment]
It is worth posting this chart:

![image](https://user-images.githubusercontent.com/12515440/198174873-c9ffeef2-0232-49f8-a8d6-4287c27292f1.png)

**Comment by snakers4** [comment]
- Remove picovoice mentions

**Comment by snakers4** [comment]
- Deprecate language classifier and number detector models, since they are not maintained anymore.

**Comment by snakers4** [comment]
# **Finally, V5 is here, 3x faster, supporting 6000+ languages!**

![image](https://github.com/snakers4/silero-vad/assets/12515440/5f581592-3035-411e-9803-44a2fda0af8e)

## **Performance and Model Size**

- 3x [faster](https://github.com/snakers4/silero-vad/wiki/Performance-Metrics) inference for TorchScript, 10% faster inference for ONNX;
- Now TorchScript is as fast as ONNX;
- Model size is 2x larger, 2MB vs. 1MB;

## **Quality**

- The VAD supports more than 6,000 languages now;
- Significanly more robust on noisy data;
- Overall 5-7% quality increase on clean data;
- Quality difference for 8 kHz and 16 kHz is negligible now;
- Quality difference for different window sizes is negligible => window size was deprecated;
- Added [benchmarks ](https://github.com/snakers4/silero-vad/wiki/Quality-Metrics) on 9 unique datasets (2 private) and one holistic multi-domain dataset;

## **Changes and deprecations**

- `window_size_samples` is deprecated - now the VAD only works with fixed size window;
- VAD now works with 8 kHz and 16 kHz sample rates, only with fixed 256 and 512 sample windows respectively;
- Slightly changed internal logic, now some context (part of previous chunk) is passed along with the current chunk;
- Sample rates that are a multiple of 16 kHz are still supported;

**Comment by snakers4** [comment]
# V5.1 - Experimental PyPI Package Release

- Experimental pip-package release;
- Community PRs to update the examples;
 
## What's Changed

* Adamnsandle by @adamnsandle in https://github.com/snakers4/silero-vad/pull/481
* Update microphone_and_webRTC_integration.py by @eltociear in https://github.com/snakers4/silero-vad/pull/475
* cpp example by @filtercodes in https://github.com/snakers4/silero-vad/pull/482
* Update Golang example to support model v5 by @streamer45 in https://github.com/snakers4/silero-vad/pull/489
* Create python-publish.yml by @adamnsandle in https://github.com/snakers4/silero-vad/pull/492
* Adamnsandle by @adamnsandle in https://github.com/snakers4/silero-vad/pull/493

## New Contributors
* @eltociear made their first contribution in https://github.com/snakers4/silero-vad/pull/475
* @filtercodes made their first contribution in https://github.com/snakers4/silero-vad/pull/482

**Full Changelog**: https://github.com/snakers4/silero-vad/compare/v5.0...v5.1

**Comment by snakers4** [comment]
# V5.1.2 - Experimental PyPI Package Release

- Minor fixes;
- Pip version update;

**Comment by MonolithFoundation** [comment]
Congrat, how to using v5 out of box

**Comment by snakers4** [comment]
A poll about export formats - https://github.com/snakers4/silero-vad/issues/706

**Comment by snakers4** [comment]
Example of VAD [ExecuTorch](https://github.com/pytorch/executorch) model usage in [C++](https://github.com/pytorch/executorch/tree/main/examples/models/silero_vad)

May be useful for some applications
